### PR TITLE
[docs] Documentation for the C++ language API.

### PR DESCRIPTION
**Pull request by AmeerHajAli**
Documentation for the C++ language API.
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ericl** [comment]
Thanks! I'll try this out.

File: doc/source/index.rst [hunk 1]
```diff
@@ -111,7 +111,74 @@ Ray provides Python and Java API. And Ray uses Tasks (functions) and Actors (Cla
             }
           }
         }
+  .. group-tab:: C++ (Experimental)
 
+    The C++ language API is in very early experimental stage.
```
Comment (ericl) [inline doc/source/index.rst:116]:
```suggestion
    The C++ Ray API is currently experimental. You can track its development `here <https://github.com/ray-project/ray/milestone/17>`__ and report issues on GitHub.
```

**Comment by AmeerHajAli** [comment]
@ericl , I made it better now by providing an example and where to put the files.
Merge at will. 

Thanks for the feedback.

**Comment by ericl** [comment]
Can you actually copy the test file so we can have a clean example that doesn't depend on gtest?

**Comment by ericl** [comment]
Closing stale PR. Please re-open if you still want to merge it.

**Pull request closed without merging**